src/golang/lib/job/errors.go

```diff
@@ -0,0 +1,74 @@
+package job
+
+import "github.com/dropbox/godropbox/errors"
+
+// JobErrorCode come from our JobManagers when they fail to properly guide
+// their a job through its proper lifecycle. Errors surfaced this way are propagated
+// outside of the python executor context. Their meaning is consistent across all
+// types of JobManagers.
+type JobErrorCode int
+
+const (
+	// System indicates an unexpected system issue that we cannot recover from.
+	System JobErrorCode = 0
+
+	// User error code indicates that the issue was the user's fault, and to surface that message
+	// to the user.
+	User = 1
+
+	// JobMissing indicates that the job manager could not find the specified job.
+	JobMissing = 2
+
+	// Noop indicates that the job manager does not have the context to make a definitive claim
+	// as to whether the job has succeeded or not. This is equivalent to saying "Do not ask me, I don't know.".
+	//
+	// For example, polling a Lambda JobManager will return this error because there is no concept of
+	// fetching a specific job's information from Lambda. The caller must figure out the job status
+	// through other means.
+	Noop = 2
+)
+
+type JobError interface {
```

Comment:
When would we have a different implementation of this interface?

Reply:
It's mainly because it needs to implement the errors interface, and so that we can `return nil` as a job error.